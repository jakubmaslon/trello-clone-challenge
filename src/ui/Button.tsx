import styled from "styled-components";

export const Button = styled.button<{ disabled?: boolean }>`
    width: 100%;
    padding: ${props => props.theme.spaces.base};
    border: none;
    background: ${props => props.theme.colors.turquoise};
    color: ${props => props.theme.colors.white};
    text-transform: uppercase;
    cursor: ${props => props.disabled ? "not-allowed" : "pointer"};
    opacity: ${props => props.disabled ? 0.5 : 1};
`;
