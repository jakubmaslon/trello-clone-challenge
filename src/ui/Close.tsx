import styled from "styled-components";

export const Close = styled.div`
    position: absolute;
    top: ${props => props.theme.spaces.base};
    right: ${props => props.theme.spaces.base};
    padding: ${props => props.theme.spaces.base};
    margin: -${props => props.theme.spaces.base};
    cursor: pointer;

    &:before {
        content: "";
        position: absolute;
        top: 50%;
        left: 0;
        right: 0;
        height: 4px;
        background: ${props => props.theme.colors.grey};
        transform: rotate(45deg) scale(0.5);
    }

    &:after {
        content: "";
        position: absolute;
        top: 50%;
        left: 0;
        right: 0;
        height: 4px;
        background: ${props => props.theme.colors.grey};
        transform: rotate(-45deg) scale(0.5);
    }

    &:hover {
        &:before {
        background: ${props => props.theme.colors.darkGrey};
    }

    &:after {
        background: ${props => props.theme.colors.darkGrey};
    }
    }
`;
