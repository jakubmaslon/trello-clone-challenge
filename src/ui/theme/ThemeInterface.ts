interface Colors {
  white: string;
}

interface Spaces {
  base: string;
}

export interface Theme {
  colors: Colors;
  spaces: Spaces;
  name: string;
}
