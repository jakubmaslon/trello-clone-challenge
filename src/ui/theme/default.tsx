import { Theme } from "./ThemeInterface";

export const theme: Theme = {
  colors: {
    white: "#FFFFFF",
    turquoise: "#3C6E71",
    grey: "#D9D9D9",
    blue: "#284B63",
    darkGrey: "#353535",
    lightGrey: "#F2F2F2",
  },
  spaces: {
    base: "16px",
    half: "8px",
  },

  name: "Default",
};
