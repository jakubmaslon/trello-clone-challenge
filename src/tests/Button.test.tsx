import * as React from "react";

import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components'

import { Button } from '../ui/Button';
import { theme } from "../ui/theme/default";

const withTheme = (element: React.ReactElement) =>
  <ThemeProvider theme={theme}>
    {element}
  </ThemeProvider>

const ButtonWTheme = withTheme(<Button />);
const ButtonDisabledWTheme = withTheme(<Button disabled />);

describe("Button component tests", () => {
  it('should render with styles for not disabled Button', () => {
    const { container: { firstChild } } = render(ButtonWTheme);

    expect(firstChild).toHaveStyleRule('border', 'none');
    expect(firstChild).toHaveStyleRule('cursor', 'pointer');
    expect(firstChild).toHaveStyleRule('opacity', '1');
  })

  it('should render with styles for disabled Button', () => {
    const { container: { firstChild } } = render(ButtonDisabledWTheme);

    expect(firstChild).toHaveStyleRule('cursor', 'not-allowed');
    expect(firstChild).toHaveStyleRule('opacity', '0.5');
  });

  it('renders correctly', () => {
    const tree = renderer.create(ButtonWTheme).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
