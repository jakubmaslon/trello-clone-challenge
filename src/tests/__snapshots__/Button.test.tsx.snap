// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests renders correctly 1`] = `
.c0 {
  width: 100%;
  padding: 1rem;
  border: none;
  background: #3C6E71;
  color: #FFFFFF;
  text-transform: uppercase;
  cursor: pointer;
  opacity: 1;
}

<button
  className="c0"
/>
`;
